py/client/pydeephaven/_session_service.py

```diff
@@ -15,7 +15,16 @@ def __init__(self, session):
 
     def connect(self) -> grpc.Channel:
         """Connects to the server and returns a gRPC channel upon success."""
-        grpc_channel = grpc.insecure_channel(":".join([self.session.host, str(self.session.port)]))
+        target = ":".join([self.session.host, str(self.session.port)])
+        if self.session.target_name_override is None:
+            options = None
+        else:
+            options = (('grpc.ssl_target_name_override', self.session.target_name_override))
```

Comment:
I think you should let the user pass in the options as I would like, rather than trying to invent each and every one.  I also set my default authority, which is actually unnecessary (I took it out just now), while trying to debug this.  If we let you use the regular options, we'll not have problems with forgetting something and need another client release.

Reply:
Makes sense, done.